boa/src/exec/new/mod.rs

```diff
@@ -23,7 +23,8 @@ impl Executable for New {
         }
         let this = Value::new_object(None);
         // Create a blank object, then set its __proto__ property to the [Constructor].prototype
-        this.set_internal_slot(INSTANCE_PROTOTYPE, func_object.get_field(PROTOTYPE));
+        this.as_object_mut().expect("this is an object").prototype =
+            func_object.get_field(PROTOTYPE);
```

Comment:
I'm guessing the `expect()` here should say the opposite, right? If it fails to unwrap, then "this was not an object".